src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,222 @@
+package duke;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.Scanner;
+import java.util.ArrayList;
+
+public class Duke {
+    public static final String LONG_LINE = "______________________________________________________________________";
+    private static final String GREETING = "Hello! I'm Duke";
+    private static final String REQUEST = "What can I do for you?";
+    private static final String EXIT = "Bye. Hope to see you again soon!";
+    private static final String ADDED_TASK = "Got it. I've added this task:";
+    private static ArrayList<Task> tasks = new ArrayList<>();
+
+    public static void printGreeting() {
+        System.out.println(LONG_LINE + System.lineSeparator() + GREETING);
+        System.out.println(REQUEST + System.lineSeparator() + LONG_LINE);
+    }
+
+    public static void insertTask() throws FileNotFoundException {
+        FileAccess.readFileContents("taskdata.txt", tasks);
+    }
+
+    public static void addTask(String input) {
+        int slashIndex = input.indexOf('/');
+        Task currentTask = null;
+        try {
+            if (input.startsWith("deadline ")) {
+                currentTask = createDeadlineTask(input, slashIndex);
+            } else if (input.startsWith("event ")) {
+                currentTask = createEventTask(input, slashIndex);
+            } else if (input.startsWith("todo ")) {
+                currentTask = createTodoTask(input);
+            } else {
+                throw new DukeException(input);
+            }
+            if (currentTask != null) {
+                tasks.add(currentTask);
+                changeFile();
+                System.out.println(LONG_LINE + System.lineSeparator() + ADDED_TASK);
+                System.out.println("  " + currentTask);
+                String plural = (tasks.size() > 1) ? "s" : "";
+                System.out.println("Now you have " + tasks.size() + " task" + plural + " in the list.");
+                System.out.println(LONG_LINE);
+            }
+        } catch (DukeException e) {
+            e.printErrorMessage();
+        }
+
+    }
+
+    public static void changeFile() {
+        try {
+            FileAccess.writeToFile("taskdata.txt", tasks);
+        } catch (IOException e) {
+            System.out.println("Something went wrong: " + e.getMessage());
+        }
+    }
+
+    private static boolean checkValid(String input, int slashIndex) {
+        boolean hasAt = input.contains(" /at ");
+        boolean hasBy = input.contains(" /by ");
+        boolean exceedLength = (slashIndex + 4) >= input.length();
+        boolean hasDeadlineTask = (slashIndex - 1) >= 9;
+        boolean hasEventTask = (slashIndex - 1) >= 6;
```

Comment:
It would be good to assign these magic numbers to a variable so that we can improve readability of the code, and understand what is the intention at first glance